src/routes/transactions/mappers/multisig-transaction.mapper.ts

```diff
@@ -0,0 +1,592 @@
+            chainId,
+            safe.address,
+          ),
+        };
+      case this.ENABLE_MODULE:
+        return <EnableModule>{
+          type: 'ENABLE_MODULE',
+          module: await this.addressInfoHelper.getOrDefault(
+            chainId,
+            dataDecoded.parameters[0]?.value,
+          ),
+        };
+      case this.DISABLE_MODULE:
+        return <DisableModule>{
+          type: 'DISABLE_MODULE',
+          module: await this.addressInfoHelper.getOrDefault(
+            chainId,
+            dataDecoded.parameters[1]?.value,
+          ),
+        };
+      case this.CHANGE_THRESHOLD:
+        return <ChangeThreshold>{
+          type: 'CHANGE_THRESHOLD',
+          threshold: dataDecoded.parameters[0]?.value,
+        };
+      case this.SET_GUARD:
+        const guardValue = dataDecoded.parameters[0]?.value;
+        if (guardValue !== this.NULL_ADDRESS) {
+          return <SetGuard>{
+            type: 'SET_GUARD',
+            guard: await this.addressInfoHelper.getOrDefault(
+              chainId,
+              guardValue,
+            ),
+          };
+        }
+        return <DeleteGuard>{
+          type: 'DELETE_GUARD',
+        };
+    }
+  }
+
+  private mapTxStatus(transaction: MultisigTransaction, safe: Safe): string {
+    if (transaction.isExecuted) {
+      return transaction.isSuccessful ? 'SUCCESS' : 'FAILED';
+    }
+    if (safe.nonce > transaction.nonce) {
+      return 'CANCELLED';
+    }
+    if (
+      this.getConfirmationsCount(transaction) <
+      this.getConfirmationsRequired(transaction, safe)
+    ) {
+      return 'AWAITING_CONFIRMATIONS';
+    }
+    return 'AWAITING_EXECUTION';
+  }
+
+  private getConfirmationsCount(transaction: MultisigTransaction): number {
+    return transaction.confirmations?.length || 0;
+  }
+
+  private getConfirmationsRequired(
+    transaction: MultisigTransaction,
+    safe: Safe,
+  ): number {
+    return transaction.confirmationsRequired ?? safe.threshold;
+  }
+
+  private getMissingSigners(
+    transaction: MultisigTransaction,
+    safe: Safe,
+    txStatus: string,
+  ): string[] {
+    console.log(transaction, safe, txStatus);
+    const confirmedOwners =
+      transaction.confirmations?.map((confirmation) => confirmation.owner) ??
+      [];
+
+    return safe.owners.filter((owner) => !confirmedOwners.includes(owner));
+  }
+
+  private mapExecutionInfo(
+    transaction: MultisigTransaction,
+    safe: Safe,
+    txStatus: string,
+  ): ExecutionInfo {
+    const executionInfo = {
+      type: 'MULTISIG',
+      nonce: transaction.nonce,
+      confirmationsRequired: this.getConfirmationsRequired(transaction, safe),
+      confirmationsSubmitted: this.getConfirmationsCount(transaction),
+    };
+
+    if (txStatus === 'AWAITING_CONFIRMATIONS') {
+      return {
+        ...executionInfo,
+        missingSigners: this.getMissingSigners(transaction, safe, txStatus),
+      };
```

Comment:
What I mean is that every time you need to return an `ExecutionInfo` you need to use the spread operator for it to be "correct" right?

I think it's preferable to return the actual instance of `ExecutionInfo` than an implementation (which can be extended of `ExecutionInfo`).

If in order to instantiate a `ExecutionInfo` we need to think about what to include or not include then I think a constructor solution is more simple and also more elegant (easier to read compared to the spread operator).

Reply:
Example:

Given the following route entities:

```typescript
export abstract class ExecutionInfo {
  @ApiProperty()
  type: string;

  protected constructor(type: string) {
    this.type = type;
  }
}

export class MultisigExecutionInfo extends ExecutionInfo {
  @ApiProperty()
  nonce: number;
  @ApiProperty()
  confirmationsRequired: number;
  @ApiProperty()
  confirmationsSubmitted: number;
  @ApiPropertyOptional()
  missingSigners: AddressInfo[] | null;

  constructor(
    nonce: number,
    confirmationsRequired: number,
    confirmationsSubmitted: number,
    missingSigners: AddressInfo[] | null,
  ) {
    super('MULTISIG');
    this.nonce = nonce;
    this.confirmationsRequired = confirmationsRequired;
    this.confirmationsSubmitted = confirmationsSubmitted;
    this.missingSigners = missingSigners;
  }
}
```

You can do the following

```typescript
private mapExecutionInfo(
    transaction: MultisigTransaction,
    safe: Safe,
    txStatus: string,
  ): ExecutionInfo {
    const missingSigners =
      txStatus === 'AWAITING_CONFIRMATIONS'
        ? this.getMissingSigners(transaction, safe)
        : null;

    return new MultisigExecutionInfo(
      transaction.nonce,
      this.getConfirmationsRequired(transaction, safe),
      this.getConfirmationsCount(transaction),
      missingSigners,
    );
  }
```

This accomplishes the following:
- You no longer need to specify a `type`. This could result in issues if other strings are applied but we want to enforce a specific type and not any `string`.
- By no longer relying on the spread operator we moved the logic on how to build this entity to the entity itself – which in my opinion avoids mistakes elsewhere.

Reply:
Alternatively you could set the type to `type: "MULTISIG"`.

However when Module transactions are introduced you'd need to have `type: "MULTISIG" | "MODULE"`. But the schema for both is quite different (module transactions only have one property: `address`.